src/hostnet/slirp.ml

```diff
@@ -1058,6 +1060,25 @@ module Make(Config: Active_config.S)(Vmnet: Sig.VMNET)(Dns_policy: Sig.DNS_POLIC
                     let ip, _ = v in
                     l @ [ip]) t.client_uuids.table []
             in 
+
+            (* check if a specific IP is requested *)
+            let preferred_ip = 
+                let uuid_bytes = Uuidm.to_bytes uuid in
+                let uuid_prefix = Bytes.sub uuid_bytes 0 (Bytes.length default_uuid_preferred_ip_prefix) in
+                if (Bytes.compare uuid_prefix default_uuid_preferred_ip_prefix) = 0 then
+                begin
+                    let uuid_suffix = Bytes.sub uuid_bytes 12 4 in
+                    let preferred_ip = Ipaddr.V4.of_bytes_exn uuid_suffix in
+                    Log.info (fun f -> f "Client requested IP %s" (Ipaddr.V4.to_string preferred_ip));
```

Comment:
Yes, I agree this would be a better approach. As discussed directly I've added an issue tracking what needs to be done to have a separate `preferred_ip` command in https://github.com/docker/vpnkit/issues/209